actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +100,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
```

Comment:
>  I settled on a successful connection means actual server connection and a protocol is chosen.

:+1: to that, but you're setting `disconnected = false` before checking the protocol.